prepare script fails: missing generate step before build

## Problem

The `prepare` script in `package.json` only runs `npm run build`, but the build depends on generated code in `src/generated/` which is gitignored. When this package is installed as a git dependency (e.g. via npm/Dependabot), the install fails because the generated files don't exist.

### Build log (from [WXYC/archive PR #11](https://github.com/WXYC/archive/pull/11))

```
✘ [ERROR] Could not resolve "../generated/models/index.js"

    src/dtos/index.ts:9:14:
      9 │ export * from '../generated/models/index.js';
        ╵               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### What happens

1. npm clones the repo — `src/generated/` is missing (gitignored)
2. `prepare` runs `npm run build` (tsup)
3. tsup tries to bundle `src/dtos/index.ts`, which imports from `../generated/models/index.js`
4. Build fails

## Fix

The `prepare` script needs to generate the TypeScript code before building:

```diff
- "prepare": "npm run build"
+ "prepare": "npm run generate:typescript && npm run build"
```

This requires `@openapitools/openapi-generator-cli` (or whichever package provides `openapi-generator-cli`) to be available in `devDependencies` so it's present when `prepare` runs.

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
   },
   "scripts": {
     "build": "tsup",
-    "prepare": "npm run build",
+    "prepare": "npm run generate:typescript && npm run build",
     "dev": "tsup --watch",
     "test": "vitest run",
     "test:watch": "vitest",

